### PR TITLE
Updated `departmentId` generation for Multiplication Projects

### PR DESCRIPTION
**Pull request by andrewmurraydavid**
## Description

Updated the EdgeDB trigger for update and insert on projects to account for the new style of department ID generation, special for the Multiplication Translation Projects.

Screenshot 2024-04-17 at 5 59 22 PM

- [x] Change the task url above to the actual Monday task
- [ ] Add/update tests if needed
- [ ] Add reviewers to this PR

### CODE REVIEW

File: dbschema/project.esdl [hunk 1]
```diff
@@ -28,19 +28,24 @@ module default {
...
-            available := (
-              <str>range_unpack(range(fa.accountNumber * 10000 + 11, fa.accountNumber * 10000 + 9999))
-              except existing
+            info := (
+              if __subject__ is MultiplicationTranslationProject
+                then (prefix := 8, startingOffset := 201)
+              else (
+                  prefix := (
+                    assert_exists(
+                      (select FundingAccount order by .accountNumber desc limit 1),
```
Comment (CarsonF) [inline dbschema/project.esdl:37]:
```suggestion
                      __subject__.primaryLocation.fundingAccount,
```

File: dbschema/project.esdl [hunk 2]
```diff
@@ -28,19 +28,24 @@ module default {
...
-            ),
-            existing := (
-              select detached Project.departmentId filter Project.primaryLocation.fundingAccount = fa
-            ),
-            available := (
-              <str>range_unpack(range(fa.accountNumber * 10000 + 11, fa.accountNumber * 10000 + 9999))
-              except existing
+            info := (
+              if __subject__ is MultiplicationTranslationProject
+                then (prefix := 8, startingOffset := 201)
```
Comment (CarsonF) [inline dbschema/project.esdl:33]: Was 80201 the right starting number?

File: src/components/project/handlers/set-department-id.handler.ts [hunk 3]
```diff
@@ -75,8 +85,9 @@ export class SetDepartmentId implements IEventHandler<SubscribedEvent> {
         CREATE (project)-[newDepartmentIdRelationship:departmentId { active: true, createdAt: datetime() }]->(:Property { createdAt: datetime(), value: departmentId })
         RETURN departmentId
         `,
-        { departmentIdPrefix: departmentIdPrefix, projectId: project.id },
+        { ...info, projectId: project.id },
       )
+      .logIt()
```
Comment (CarsonF) [inline src/components/project/handlers/set-department-id.handler.ts:90]:
```suggestion
```

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/components/project/handlers/set-department-id.handler.ts (hunk 3 above)
Comment (andrewmurraydavid) [reply src/components/project/handlers/set-department-id.handler.ts:90]: ugh, derp

File: dbschema/project.esdl (hunk 2 above)
Comment (andrewmurraydavid) [reply dbschema/project.esdl:33]: Just for github visibility: a comment on the story specifies 80201 would be the first: 
> Restrict 80200 and below (?)

File: dbschema/project.esdl (hunk 1 above)
Comment (andrewmurraydavid) [reply dbschema/project.esdl:37]: another derp. good catch!

**Review by CarsonF — APPROVED** [review]
LGTM! I'll get a migration and merge shortly

**Pull request merged**